### ISSUE TITLE
How to use digest authentication?

### ISSUE THREAD

**Issue by AcroManiac** [Microsoft/cpprestsdk#1044]
Hello,

I am trying to make http request for Wowza Streaming Engine RESTful API server which is using digest authentication.

I use web::credentials in http_client_config object but it works for basic authorization only. In digest mode my request returns HTTP error 401 ("Unauthorized").

Could somebody tell how to modify my code to use http request with digest authentication in cpprest SDK?

        // Create authorization for HTTP client
        http_client_config  config;
        config.set_credentials(web::credentials(
            utility::conversions::to_string_t("admin"),
            utility::conversions::to_string_t("******")));

        // Create http_client to send the request
        http_client  client(sServerPath, config);
        http_request request(bStreamingOn ? methods::POST : methods::PUT);
        request.headers().set_content_type(U("application/json; charset=utf-8"));
        request.headers().add(U("Accept"), U("application/json; charset=utf-8"));   // get result in JSON format
        request.set_request_uri(sCommandPath);
        request.set_body(sBodyData);

        // Start the request to Wowza REST Api
        client.request(request)
                .then([=](http_response response) mutable
                {
                    Logger.LogMessage(
                                NLogger2::ePriorityLevelDebug,
                                std::string("Received response status code"),
                                "CTaskBuilderSingleton::RecordMediaStream()",
                                std::to_string(response.status_code()));
                    return response.extract_json();
                })
                .then([=](json::value result) mutable
                {
                    Logger.LogMessage(
                                NLogger2::ePriorityLevelDebug,
                                std::string("Wowza response"),
                                "CTaskBuilderSingleton::RecordMediaStream()",
                                result.serialize());
                })
                .then([=](pplx::task<void> t) mutable
                {
                    try
                    {
                        t.get();
                    }
                    catch (...)
                    {
                        Logger.LogMessage(
                                    NLogger2::ePriorityLevelError,
                                    std::string("Some error occured while sending Wowza recording command"),
                                    "CTaskBuilderSingleton::RecordMediaStream()");
                    }
                })
        ;

**Comment by mandarsudame** [comment]
Were you able to find a solution to this problem?
I am facing a similar issue with bearer token auth.

**Comment by AcroManiac** [comment]
No, unfortunately not. I am using it with basic authentication only.

**Comment by deeringc** [comment]
This should work out of the box on windows (building on top of WinHttp) but isn't implemented on other platforms (which use the asio client)
